rust/psibase_macros/psibase-macros-lib/src/service_macro/mod.rs

```diff
@@ -92,7 +92,10 @@ fn process_mod(
         let mut action_fns: Vec<usize> = Vec::new();
         let mut non_action_fns: Vec<usize> = Vec::new();
         let mut event_fns: HashMap<EventType, Vec<usize>> = HashMap::new();
-        check_for_pre_action(&mut pre_action_info, items);
+        let pa_ret = check_for_pre_action(&mut pre_action_info, items);
+        if pa_ret.is_err() {
+            return pa_ret.err().unwrap();
+        }
```

Comment:
This returning of an Err and then having to unwrap it this way seems like it must be too much, but I can't find a simpler/more elegant way to do these 4 lines. @swatanabe @James-Mart